TicketsDemo.Domain/DefaultImplementations/TicketServiceLoggingDecorator.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Domain.Interfaces;
+
+namespace TicketsDemo.Domain.DefaultImplementations
+{
+    public class TicketServiceLoggingDecorator: ITicketService
+    {
+        public ITicketService _decoratedObject;
+        public ILogger _logger;
+
+        public TicketServiceLoggingDecorator(ITicketService decoratedObject,  ILogger logger)
+        {
+            _decoratedObject = decoratedObject;
+            _logger = logger?? new FileLogger();
```

Comment:
i'd rather remove the inline default logger creation.
Although it is not the "mistake" but rather my personal preference.

The bad thing about having dependencies between concrete classes is that it might cause circular dependencies between classes (easy to track and resolve) or assemblies (much harder to resolve).